improc/simulator.py

```diff
@@ -453,6 +501,331 @@ def get_star_flux_values(self):
+        elif isinstance(exp_scale_ratios, (int, float)):
+            exp_scale_ratios = np.full(number, exp_scale_ratios)
+        self.galaxy_exp_scale_ratios = np.append(self.galaxy_exp_scale_ratios, exp_scale_ratios)
+
+        if sersic_scale_ratios is None:
+            sersic_scale_ratios = rng.uniform(0.5, 2, number) * widths
+        elif isinstance(sersic_scale_ratios, (int, float)):
+            sersic_scale_ratios = np.full(number, sersic_scale_ratios)
+        self.galaxy_sersic_scale_ratios = np.append(self.galaxy_sersic_scale_ratios, sersic_scale_ratios)
+
+        if sersic_flux_ratios is None:
+            sersic_flux_ratios = 10 ** rng.uniform(-2, -1, number)
+        elif isinstance(sersic_flux_ratios, (int, float)):
+            sersic_flux_ratios = np.full(number, sersic_flux_ratios)
+        self.galaxy_sersic_flux_ratios = np.append(self.galaxy_sersic_flux_ratios, sersic_flux_ratios)
+
+        if cos_is is None:
+            cos_is = rng.uniform(0.1, 1, number)
+        elif isinstance(cos_is, (int, float)):
+            cos_is = np.full(number, cos_is)
+        self.galaxy_cos_is = np.append(self.galaxy_cos_is, cos_is)
+
+        if rotations is None:
+            rotations = rng.uniform(0, 360, number)
+        elif isinstance(rotations, (int, float)):
+            rotations = np.full(number, rotations)
+        self.galaxy_rotations = np.append(self.galaxy_rotations, rotations)
+
+    def remove_galaxies(self, number=1):
+        """Remove the latest few galaxies (default is only one) from the galaxy field. """
+        if number > 0:
+            self.galaxy_mean_fluxes = self.galaxy_mean_fluxes[:-number]
+            self.galaxy_mean_x_pos = self.galaxy_mean_x_pos[:-number]
+            self.galaxy_mean_y_pos = self.galaxy_mean_y_pos[:-number]
+            self.galaxy_width_values = self.galaxy_width_values[:-number]
+            self.galaxy_cos_is = self.galaxy_cos_is[:-number]
+            self.galaxy_rotations = self.galaxy_rotations[:-number]
+            self.galaxy_sersic_flux_ratios = self.galaxy_sersic_flux_ratios[:-number]
+            self.galaxy_sersic_scale_ratios = self.galaxy_sersic_scale_ratios[:-number]
+            self.galaxy_exp_scale_ratios = self.galaxy_exp_scale_ratios[:-number]
+
+    def get_galaxy_x_values(self):
+        """
+        Return the positions of the galaxies (in pixel coordinates)
+        after possibly applying small astrometric shifts (e.g., scintillations)
+        to the mean galaxy positions.
+
+        """
+        x = self.galaxy_mean_x_pos.copy()
+        if self.galaxy_position_std is not None:
+            x += np.random.normal(0, self.galaxy_position_std, len(x))
+
+        return x
+
+    def get_galaxy_y_values(self):
+        """
+        Return the positions of the galaxies (in pixel coordinates)
+        after possibly applying small astrometric shifts (e.g., scintillations)
+        to the mean galaxy positions.
```

Comment:
Aside: another (more painful to implement) use of shifting positions is differential chromatic refraction.